### PR TITLE
Cutter docs update

### LINKED ISSUE #744: Ideas for improving documentation
Our new documentation lies here: https://cutter.re/docs/
Many things are lacking, and some are not very well explained.
It would be awesome if someone could complete the documentation, providing screenshots, etc

**edited by @itayc0hen :**

We are talking about creating a clear and easy-to-follow documentation for Cutter.

Including but not limited -

- Compilation tutorials
- Widgets description (with screenshots for better understanding)
- Shortcuts
- Settings explanation
- Common use cases
- Common Issues
- Cutter Limits
- How to contribute (opening issues, opening Pull Requests)
- Contributing guidelines (copy content from CONTRIBUTING.md)

As you can see on our [website](https://radareorg.github.io/cutter), we already have some pages such as 
*Building, Common errors, Overview, Shortcuts* that we want to improve as well as adding more.

I'd suggest you take inspiration from the Documentation of BinaryNinja: https://docs.binary.ninja/getting-started.html

### PR DESCRIPTION
**Pull request by surendrajat**
Some updates to the Cutter docs.

**Your checklist for this pull request**
- [x] I've read the [guidelines for contributing](https://cutter.re/docs/developers-docs/first-time.html) to this repository
- [x] I made sure to follow the project's [coding style](https://cutter.re/docs/developers-docs.html)
- [x] I've updated the [documentation](https://cutter.re/docs/user-docs.html) with the relevant information (if needed)

**Detailed description**
This PR has the following:
- Move building errors from common errors to building page and move rest under user-docs/common-issues
- Title capitalization (IDK I'm obsessed with it xD)
- Updated troubleshooting part with git submodule related error
- Apply Grammarly suggestions on all the docs

**Test plan (required)**
Build and run docs.

**Closing issues**
Partially close$ #744

### CODE REVIEW

**Comment by Surendrajat** [comment]
I've added a widget page with few moving screenshots(WIP). Let me know what you think.

**Comment by karliss** [comment]
I would be much happier if documentation was moved to separate repository before adding a bunch of screenshots for all the widgets and dialogs. Those could quickly cause bloat.

**Comment by XVilka** [comment]
It can be made a submodule to keep the old structure.

**Comment by Surendrajat** [comment]
I see #2056 open for just this. If it's done I can move things there.
As for submodule, I think it'll add additional commits to this repo on each update to docs, which will probably be better if avoided. How about splitting API docs and other docs and leave API docs here?

**Comment by ITAYC0HEN** [comment]
You might want to consider adding the `:menuselection:` and `:guilabel:` to the docs as part of this or another PR. @karliss found these features just recently

https://www.sphinx-doc.org/en/master/usage/restructuredtext/roles.html

**Comment by Surendrajat** [comment]
@karliss I've removed the widget intro part and will open another PR to continue work on that until we decide if to split repo or not.

File: docs/source/index.rst [hunk 1]
```diff
@@ -44,7 +44,7 @@ Want to help the project?
 -------------------------
 
 If you want to contribute to Cutter, take a look
-`here <contributing>`_
+`here <contributing.html>`_
```
Comment (karliss) [inline docs/source/index.rst:47]: Please don't do that. Sphinx can't check if link is valid if you use html urls. Use the sphinx/rst builtin mechanisms for referencing documents or headers.

Comment (Surendrajat) [reply docs/source/index.rst:47]: Not sure how it's working in docs but not locally. Will check again and fix.

Comment (karliss) [reply docs/source/index.rst:47]: If it doesn't work for you locally version without extension is likely also wrong.

Comment (ITAYC0HEN) [reply docs/source/index.rst:47]: Also, I prefer to write the full description and not "here".

For example "take a look at our contribution guidelines" or "take a look at our guidelines for new contributors" or "read our tutorials for contributors" etc

File: docs/source/building.rst [hunk 2]
```diff
@@ -307,3 +307,43 @@ The workaround is to either add the `--disable-new-dtags` linker flag when compi
 
    cmake -DCMAKE_EXE_LINKER_FLAGS="-Wl,--disable-new-dtags"  ..
 
+* **r_*.h: No such file or directory**
+
+    r_util/r_annotated_code.h: No such file or directory
+
+If you face an error where some header file starting with ``r_`` is missing, you should check the **radare2** submodule and
+make sure it is in the sync with upstream **Cutter** repo. Simply run:
```
Comment (ITAYC0HEN) [inline docs/source/building.rst:315]:
```suggestion
make sure it is in sync with upstream **Cutter** repo. Simply run:
```

or

```suggestion
make sure it is synced with upstream **Cutter** repo. Simply run:
```

File: docs/source/building.rst [hunk 3]
```diff
@@ -307,3 +307,43 @@ The workaround is to either add the `--disable-new-dtags` linker flag when compi
...
+    git sumodule update
+
+* **r_core development package not found**
+
+If you installed radare2 and still encounter this error, it could be that your
+``PATH`` environment variable is set improperly (doesn’t contain
+``/usr/local/bin``). That can be, for example, due to ``Qt Creator.app``
+being copied over to ``/Applications``. To fix this, append
+``:/usr/local/bin`` to the ``PATH`` variable within the *Build
+Environment* section in Qt Creator. See the screenshot below should you
```
Comment (ITAYC0HEN) [inline docs/source/building.rst:328]: Only Qt Creator?

File: docs/source/building.rst [hunk 4]
```diff
@@ -307,3 +307,43 @@ The workaround is to either add the `--disable-new-dtags` linker flag when compi
...
+encounter any problems.
+
+On macOS, you can also try:
+
+-  ``PKG_CONFIG_PATH=$HOME/bin/prefix/radare2/lib/pkgconfig qmake``
+-  ``PKG_CONFIG_PATH=$HOME/cutter/radare2/pkgcfg qmake`` (for modern
+   version and if radare2 was installed like the submodule)
```
Comment (ITAYC0HEN) [inline docs/source/building.rst:335]: What is this "modern version"? Maybe should be more specific (e.g for versions newer than 1.2.3.4, ....)

Comment (ITAYC0HEN) [reply docs/source/building.rst:335]: Would rephrase "was installed like the submodule" because the meaning isn't that clear

File: docs/source/building.rst [hunk 5]
```diff
@@ -307,3 +307,43 @@ The workaround is to either add the `--disable-new-dtags` linker flag when compi
...
+
+.. image:: images/cutter_path_settings.png
+
+Or radare2 libs could be installed to ``/usr/lib/pkgconfig/``, so you
```
Comment (ITAYC0HEN) [inline docs/source/building.rst:339]: I'd rephrase the opening "or" and the "could be" here. I am not sure about the meaning of the sentence. Maybe "If radare2 libs can be installed to..." or "It is possible that radare2 libs..."

File: docs/source/building.rst [hunk 6]
```diff
@@ -307,3 +307,43 @@ The workaround is to either add the `--disable-new-dtags` linker flag when compi
...
+can add variable ``PKG_CONFIG_PATH`` with value ``/usr/lib/pkgconfig/``
```
Comment (ITAYC0HEN) [inline docs/source/building.rst:340]:
```suggestion
can add a variable ``PKG_CONFIG_PATH`` with the value ``/usr/lib/pkgconfig/``
```

File: docs/source/building.rst [hunk 7]
```diff
@@ -307,3 +307,43 @@ The workaround is to either add the `--disable-new-dtags` linker flag when compi
...
+
+* **macOS libjpeg error**
+
+On macOS, QT5 apps fail to build on QtCreator if you have the ``libjpeg``
```
Comment (ITAYC0HEN) [inline docs/source/building.rst:344]:
```suggestion
On macOS, QT5 apps fail to build on QtCreator if you have the ``libjpeg``
```
```suggestion
On macOS, Qt5 apps fail to build on QtCreator if you have the ``libjpeg``
```

File: docs/source/contributing/code/development-guidelines.rst [hunk 8]
```diff
@@ -32,24 +32,24 @@ There are multiple ways to call a radare2 command:
...
 
-The command then returns its output. This should be used when a command doesn't have output or the output should be handled as-is. If possible, using the json variation with ``cmdj`` is always preferred.
+The command then returns its output. This should be used when a command doesn't have output or the output should be handled as-is. If possible, using the JSON variation with ``cmdj`` is always preferred.
   
 - ``CutterCore::cmdRawAt(<command>, <address>)`` - Executes a single radare2 command in a given address and returns the output. This helps avoiding weird strings concatenation like ``cmd("ph " + hash + " @ " + QString::num(address))``.
   
 - ``CutterCore::cmd()`` - *(Discouraged)* Only use it when ``cmdj`` or ``cmdRaw`` cannot be used. This is used for complex commands using concatenation of several commands (``px 5; pd 7; afl;``), for grepping (``pd 5~call``). for commands inside commands (``?e `afn.```) and so on.
-  This is also used when the output is complex and does not parsed correctly in ``cmdRaw``.
+  This is also used when the output is complex and does not get parsed correctly in ``cmdRaw``.
```
Comment (ITAYC0HEN) [inline docs/source/contributing/code/development-guidelines.rst:40]:
```suggestion
  This is also used when the output is complex and is not parsed correctly in ``cmdRaw``.
```

File: docs/source/contributing/code/development-guidelines.rst [hunk 9]
```diff
@@ -190,22 +190,22 @@ To connect a signal to a slot, this is the preferred syntax:
...
+Updating the Git Submodules
+~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
 Git submodules play a major part in Cutter. This, because Cutter is powered
 by radare2, its parent project, and it tries to stay up-to-date with its
 recent version, which allows us to implement new features, and enjoy bug
 fixes and performance improvements on radare2. Often, we need to update
-the radare2 submodule or others, in order to push the most recent
+the radare2 submodule or the others, to push the most recent
 version of them to Cutter.
```
Comment (ITAYC0HEN) [inline docs/source/contributing/code/development-guidelines.rst:209]:
```suggestion
the radare2 submodule or the others, to push their most recent
version to Cutter.
```

File: docs/source/user-docs/command-line.rst [hunk 10]
```diff
@@ -1,4 +1,4 @@
-Command line options
+CommandLine Options
```
Comment (ITAYC0HEN) [inline docs/source/user-docs/command-line.rst:1]:
```suggestion
Command-Line Options
```
or
```suggestion
Command-line Options
```
or
```suggestion
Command Line Options
```

File: docs/source/user-docs/common-issues.rst [hunk 11]
```diff
@@ -0,0 +1,19 @@
...
+=============
+
+This page lists common issues encountered by users.
+
+AppImage Crashes
+----------------
+
+If the Linux AppImage binary crashes upon startup, make sure your
+``LD_LIBRARY_PATH`` environment variable is empty.
+`#579 <https://github.com/radareorg/cutter/issues/579>`__
```
Comment (ITAYC0HEN) [inline docs/source/user-docs/common-issues.rst:11]: Seems like this weird number is just thrown there out of no-where. Try to add an opening sentence to introduce the number. "For a detailed explanation see issue #1234"

File: docs/source/user-docs/menus/disassembly-context-menu.rst [hunk 12]
```diff
@@ -73,23 +73,23 @@ Re-type Local Variables
...
-Delete a flag
+Delete a Flag
 ----------------------------------------
 **Description:**   Delete the flag at the current location.
 
-**Steps:** Right click on a location with a flag and choose ``Delete flag``.  
+**Steps:** Right-click on a location with a flag and choose ``Delete flag``.  
 
-Undefine a function
+UnDefine a Function
```
Comment (ITAYC0HEN) [inline docs/source/user-docs/menus/disassembly-context-menu.rst:92]:
```suggestion
Undefine a Function
```

The capital `D` is not needed. Undefine, Unbelievable , Undo, Unconventional, Unfamiliar

File: docs/source/user-docs/menus/disassembly-context-menu/debug-context-menu.rst [hunk 13]
```diff
@@ -1,14 +1,14 @@
...
+Continue Until Line
 ----------------------------------------
 **Description:** Continue the execution of the program until reached the selected offset. The program is not guaranteed to ever reach this address and will keep running until exited or until reached another breakpoint. If other breakpoints hit before reaching this line, they will be triggered and pause the execution. *This option is only available on Debug or Emulation modes*.      
 
 **Steps:** While in Debug or Emulation modes, right-click and address and choose ``Debug -> Continue until line``.  
 
 Set Program Counter (PC)
 ----------------------------------------
-**Description:** Set the Program Counter of the debuggee to the current offset. For example, on Intel 64bit program, Cutter will set the value of the RIP register to the curren address.  *This option is only available on Debug or Emulation modes*.  
+**Description:** Set the Program Counter of the debugger to the current offset. For example, on a Intel 64bit program, Cutter will set the value of the RIP register to the current address.  *This option is only available on Debug or Emulation modes*.
```
Comment (ITAYC0HEN) [inline docs/source/user-docs/menus/disassembly-context-menu/debug-context-menu.rst:12]: The "debuggee" here is intentional -> https://wikidiff.com/debugger/debuggee

**Review by ITAYC0HEN — CHANGES_REQUESTED** [review]
Fantastic work!

File: docs/source/building.rst (hunk 3 above)
Comment (Surendrajat) [reply docs/source/building.rst:328]: I copied it from common-errors :(

**Comment by ITAYC0HEN** [comment]
is there anything left to do?

**Comment by Surendrajat** [comment]
> is there anything left to do?

Yes. Finishing right away. :P

**Review by ITAYC0HEN — APPROVED** [review]
Looks good! I love this pr :)
